src/coinjoin/client.h

```diff
@@ -138,7 +138,7 @@ class CoinJoinWalletManager {
 class CCoinJoinClientSession : public CCoinJoinBaseSession
 {
 private:
-    CWallet& m_wallet;
+    const std::shared_ptr<CWallet> m_wallet;
```

Comment:
Alternatively,
if CCoinJoinClientSession and CCoinJoinClientManager is now an owner of CWallet, smart pointer still can be removed from `CTransactionBuilder`.

Reply:
so, with this PR we actually have guarantee that CWallet won't be freed during any CCoinJoinClientSession and CCoinJoinClientManager operation. Let's do a final touch then: 4f731914941564fcd41972e33dab92e5b0d8e011 to simplify code a bit further